phpunit/functional/DBmysqlIteratorTest.php

```diff
@@ -1622,35 +1600,28 @@ protected function requestArgsProvider(): iterable
         ];
     }
 
-    /**
-     * @dataProvider requestArgsProvider
-     */
+    #[DataProvider('requestArgsProvider')]
     public function testConvertOldRequestArgsToCriteria(array $params, array $expected, string $sql): void
     {
-        $this->mockGenerator->orphanize('__construct');
-        $db = new \mock\DBMysql();
-
+        $db = $this->getMockBuilder('DBMysql')
+            ->disableOriginalConstructor()
+             ->getMock();
         $iterator = new \DBmysqlIterator($db);
 
-        $result = null;
-        $this->when(
-            function () use ($iterator, $params, &$result) {
-                $result = $this->callPrivateMethod($iterator, 'convertOldRequestArgsToCriteria', $params, 'test');
-            }
-        )
-         ->error
-         ->withMessage('The `test()` method signature changed. Its previous signature is deprecated.')
-         ->withType(E_USER_DEPRECATED)
-         ->exists();
+        $result = $this->callPrivateMethod($iterator, 'convertOldRequestArgsToCriteria', $params, 'test');
 
-        $this->array($result)->isEqualTo($expected);
+        $this->hasPhpLogRecordThatContains(
+            'The `test()` method signature changed. Its previous signature is deprecated.',
+            LogLevel::INFO
+        );
+        $this->assertEquals($expected, $result);
     }
 
-    /**
-     * @dataProvider requestArgsProvider
-     */
+    /*
+    #[DataProvider('requestArgsProvider')]
     public function testExecuteWithOldSignature(array $params, array $expected, string $sql): void
     {
+        //PHPUnit cannot mack native functions.
```

Comment:
I did not understand why the need of a mock here; anyway not possible to mock `mysqli_result` with phpunit.

Reply:
Seems useless, I removed it.